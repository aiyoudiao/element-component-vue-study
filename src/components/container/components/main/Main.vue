<script lang="ts">
export default {
  name: 'ElMain',
}
</script>

<script setup lang="ts"></script>

<template>
  <main class="el-main">
    <slot />
  </main>
</template>

<style lang="scss">
@import '../../../../theme/src/main.scss'
</style>
