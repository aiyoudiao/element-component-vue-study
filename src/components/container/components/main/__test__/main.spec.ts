import { mount } from '@vue/test-utils'
import Main from '../Main.vue'
describe('Main.vue', () => {
  it('Main snapshot', () => {
    const wrapper = mount(Main)
    expect(wrapper.element).toMatchSnapshot()
  })
  it('Main render', () => {
    const wrapper = mount(Main)
    expect(wrapper.classes()).toContain('el-main')
  })
  it('Main slot', () => {
    const wrapper = mount(Main, {
      slots: {
        default: () => 'hello world',
      },
    })
    expect(wrapper.text()).toBe('hello world')
  })
})

