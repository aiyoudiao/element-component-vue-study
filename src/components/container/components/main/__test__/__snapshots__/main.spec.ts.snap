// Vitest Snapshot v1

exports[`Main.vue > Main snapshot 1`] = `
<main
  class="el-main"
>
  
  
</main>
`;
